test/TesterInternal/StorageTests/SQLAdapter/temp/TestEnvironmentInformation.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace UnitTests.StorageTests.SQLAdapter.temp
+{
+    /// <summary>
+    /// Enumeration of the supported test environment configurations.
+    /// </summary>
+    public enum SupportedTestConfigurations
```

Comment:
@jdom I moved the discussion here.

I think we are in on the same page here. I don't know what would be a suitable arrangement. Some questions we could ponder:

a. How many environments there would need to be enumerated?
b. What kind of testing would we support both for Orleans project and perhaps other people running Orleans test builds in their rig?

I use connection strings as an example here, but they could be any other operation that needs to be performed. What would be a suitable way to pick the right connection string depending on the environment? I can think of having them in a configuration file and then supply a right one (or transform one, say, with XDT) to be read when the tests start. But it still seems one needs to have them somewhere dependent on environment. A tacit assumption here is that on Windows one has VS and hence LocalDb installed, likely also Azure Storage Emulator and their default connection strings work. If they don't work, then either the precondition or fixture check fail and tests aren't run. This can happen regardless of the test environment detection, but the connection strings need to be picked from somewhere.

Then one issue is that there might be need for environment dependent setup (like setting environment variables on hosted LocalDb, but this particular need might not be a real one).

What would be a fallback and how would it work? Would the fallback be "development environment" and then detect if LocalDb (SQL Server), MySQL, ZooKeeper and Azure Storage Emulator are installed using default connection strings and trying to connect to them? That might be a sensible fallback if all else fails, but how to choose the necessary settings and perform the necessary setup on other environments? I'm not sure if there is diverge now and if not, should some precaution to be taken to prepare for likely situations.

(Also, it may be evident, but the code in the spike isn't that polished.)